oneanddone/tests/functional/tests/test_task_assignment.py

```diff
@@ -0,0 +1,27 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from pages.home import HomePage
+
+
+class TestAvailableTasks():
+
+    @pytest.mark.nondestructive
+    def test_assigned_task_is_not_available(self, base_url, selenium, assigned_task, new_user):
+        home_page = HomePage(selenium, base_url).open()
+        home_page.login(new_user)
+        available_tasks_page = home_page.click_available_tasks()
+        home_page.search_for_task(assigned_task['task'].name)
+        assert len(available_tasks_page.available_tasks) == 0
+        assert assigned_task['task'].is_available_to_user(assigned_task['user']) == True
```

Comment:
The second assert is not checking anything on the front-end or anything that is a consequence of interacting with the front-end, it's more of a unit test and should be covered elsewhere accordingly. If you want to check that the task is available via the front-end you would need to login as the assigned user and verify that you can find the task.